src/policy/space.rs

```diff
@@ -243,8 +243,13 @@ pub trait Space<VM: VMBinding>: 'static + SFT + Sync + Downcast {
 
         trace!("Polling ..");
 
-        if allow_poll && VM::VMActivePlan::global().poll(false, self.as_space()) {
```

Comment:
This is now dead code. Should be removed
```Rust
let allow_poll = unsafe { VM::VMActivePlan::is_mutator(tls) }
            && VM::VMActivePlan::global().is_initialized();
```

Reply:
It is not actually dead code. It is still used in the `else` block.

Reply:
I will refactor this part to make it cleaner.

Reply:
Ah, sorry, I missed that.

Reply:
Refactored the code a bit, and added some comments to make it clearer.

Reply:
You can take a look and let me know if it is better now (if so, also remove the change request so we can merge this when tests pass).

Reply:
Thanks Yi. It's much clearer now.